app/models/project/storage.rb

```diff
@@ -0,0 +1,109 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+module Project::Storage
+  def self.included(base)
+    base.send :extend, StorageMethods
+    base.send :include, ModelMethods
+  end
+
+  module ModelMethods
+    ##
+    # Count required disk storage for this project.
+    # Returns a hash of the form:
+    # 'total' => Total required disk space for this project
+    # 'modules' => Hash of localized keys and required space for this module
```

Comment:
Just some nit picking about words: Wouldn't **localized** keys be keys that are already localized into the correct language? Don't you mean *localization keys*?